Packages/tests/HardwareBasic/UTF_TestPulseAndTPDuringDAQ.ipf

```diff
@@ -1271,3 +1271,41 @@ static Function TPDuringDAQwithPS_REENTRY([str])
 
 	CHECK_EQUAL_WAVES(stimScale, {tpAmp, daGain, NaN, NaN, NaN, NaN, NaN, NaN, NaN}, mode = WAVE_DATA)
 End
+
+// UTF_TD_GENERATOR DeviceNameGeneratorMD1
+static Function TPZerosDAC([string str])
+	STRUCT DAQSettings s
+	InitDAQSettingsFromString(s, "MD1_RA1_I0_L0_BKG1_TP1"                + \
+								 "__HS0_DA0_AD0_CM:IC:_ST:StimulusSetA_DA_0:")
+
+	AcquireData_NG(s, str)
+
+	CtrlNamedBackGround StopTPAfterFiveSeconds, start=(ticks + TP_DURATION_S * 60), period=1, proc=StopTPAfterFiveSeconds_IGNORE
+End
+
+static Function TPZerosDAC_REENTRY([string str])
+
+	variable deviceID, hardwareType, sweepNo, DAC
+
+	CHECK_EQUAL_VAR(ROVar(GetDataAcqRunMode(str)), DAQ_NOT_RUNNING)
+	CHECK_EQUAL_VAR(ROVar(GetTestpulseRunMode(str)), TEST_PULSE_NOT_RUNNING)
+
+	sweepNo = AFH_GetLastSweepAcquired(str)
+	CHECK_EQUAL_VAR(sweepNo, NaN)
+
+	WAVE numericalValues = GetLBNumericalValues(str)
+
+	WAVE/Z DACs = GetLastSetting(numericalValues, sweepNo, "DAC", TEST_PULSE_MODE)
+	CHECK_WAVE(DACs, NUMERIC_WAVE)
+	WAVE/Z indizes = FindIndizes(DACs, prop = PROP_NON_EMPTY)
+	CHECK_WAVE(indizes, NUMERIC_WAVE)
+	CHECK_EQUAL_VAR(DimSize(indizes, ROWS), 1)
+
+	DAC = DACs[indizes[0]]
+	CHECK_GE_VAR(DAC, 0)
+
+	deviceID = ROVar(GetDAQDeviceID(str))
+	hardwareType = GetHardwareType(str)
+
+	CHECK_LE_VAR(HW_ReadADC(hardwareType, deviceID, DAC), 0.01)
```

Comment:
I think that is not correct, because you have to get the ADC channel that is associated to the DAC channel and call ReadADC with the ADC channel.

Reply:
Will fix. Is also wrong for the other new test.

Reply:
Done. I've also made the code shorter and are now testing for SD and MD.